looker/resource_looker_user_attribute_group.go

```diff
@@ -0,0 +1,196 @@
+		}
+		userAttrGroupId += *userAttrGroup.Id
+
+		tflog.Info(ctx, "CREATE", map[string]interface{}{
+			"groupId": *userAttrGroup.GroupId,
+			"attrId":  *userAttrGroup.UserAttributeId,
+			"value":   *userAttrGroup.Value,
+		})
+	}
+	d.SetId(userAttrGroupId)
+
+	return resourceUserAttributeGroupRead(ctx, d, c)
+}
+
+func resourceUserAttributeGroupRead(ctx context.Context, d *schema.ResourceData, c interface{}) diag.Diagnostics {
+	api := c.(*sdk.LookerSDK)
+
+	userAttrGroups, err := api.AllUserAttributeGroupValues(
+		d.Get("user_attribute_id").(string),
+		"",
+		nil,
+	)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	result := multierror.Append(
+		d.Set("user_attribute_id", userAttrGroups[0].UserAttributeId), // all share the same user attribute, therefore can just the first
+		d.Set("group_values", buildGroupValuesMap(ctx, d, userAttrGroups)),
+	)
+
+	return diag.FromErr(result.ErrorOrNil())
+}
+
+func resourceUserAttributeGroupUpdate(ctx context.Context, d *schema.ResourceData, c interface{}) diag.Diagnostics {
+	api := c.(*sdk.LookerSDK)
+
+	groupValues := d.Get("group_values").([]interface{})
+	for _, groupValue := range groupValues {
+		groupValueMap := groupValue.(map[string]interface{})
+		usrAttrGrp, err := api.UpdateUserAttributeGroupValue(
+			groupValueMap["group_id"].(string),
+			d.Get("user_attribute_id").(string),
+			sdk.UserAttributeGroupValue{
+				GroupId:         conv.PString(groupValueMap["group_id"].(string)),
+				UserAttributeId: conv.PString(d.Get("user_attribute_id").(string)),
+				Value:           conv.PString(groupValueMap["value"].(string)),
+			},
+			nil,
+		)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		tflog.Info(ctx, "UPDATE", map[string]interface{}{
+			"groupId": *usrAttrGrp.GroupId,
+			"attrId":  *usrAttrGrp.UserAttributeId,
+			"value":   *usrAttrGrp.Value,
+		})
+	}
+
+	return resourceUserAttributeGroupRead(ctx, d, c)
+}
+
+func resourceUserAttributeGroupDelete(ctx context.Context, d *schema.ResourceData, c interface{}) diag.Diagnostics {
+	api := c.(*sdk.LookerSDK)
+
+	groupValues := d.Get("group_values").([]interface{})
+	for _, groupValue := range groupValues {
+		groupValueMap := groupValue.(map[string]interface{})
+		err := api.DeleteUserAttributeGroupValue(
+			groupValueMap["group_id"].(string),
+			d.Get("user_attribute_id").(string),
+			nil,
+		)
+		if err != nil && !errors.Is(err, io.EOF) {
+			return diag.FromErr(err)
+		}
+	}
+
+	return nil
+}
+
+func buildGroupValuesMap(ctx context.Context, d *schema.ResourceData, userAttrGroups []sdk.UserAttributeGroupValue) []map[string]interface{} {
+	var groupValuesMap []map[string]interface{}
+
+	groupValues := d.Get("group_values").([]interface{})
+	for i, userAttrGroup := range userAttrGroups {
+		groupid := *userAttrGroup.GroupId
+		value := *userAttrGroup.Value
+
+		if *userAttrGroup.ValueIsHidden {
+			value = groupValues[i].(map[string]interface{})["value"].(string)
```

Comment:
Is the order of the groups in `groupValues` the same as the order of `userAttrGroups` returned from the API? Just wondering to make sure the value taken from the state here matches up with the right `groupid`. To be safe you could also override the `groupid` here?

Reply:
Actually just thinking, if the user attribute value is hidden, then we can't inspect what comes back from the API and the state can stay the same? Seen at the API doesn't return the `Source` of the value, is there a way to know if it's been deleted/changed or not?